packages/mobile/src/harmony-native/components/input/TextInput/TextInput.tsx

```diff
@@ -140,11 +140,19 @@ export const TextInput = forwardRef(
     const hasValue = characterCount > 0
 
     // Hide the label when requested or when the size is set to small
-    const shouldShowLabel = !hideLabel && size !== TextInputSize.SMALL
+    const shouldShowLabel =
+      !hideLabel &&
+      size !== TextInputSize.SMALL &&
+      size !== TextInputSize.EXTRA_SMALL
     const labelText = required ? `${labelProp} *` : labelProp
     const placeholderText =
       required && hideLabel ? `${placeholder} *` : placeholder
-    const helperTextSize: TextSize = size === TextInputSize.SMALL ? 'xs' : 's'
+    const helperTextSize: TextSize =
+      size === TextInputSize.EXTRA_SMALL
+        ? 'xs'
+        : size === TextInputSize.SMALL
+          ? 'xs'
+          : 's'
```

Comment:
too much nesting imo

Reply:
fair, but i feel like it's still easier to read than pulling out a constant somewhere further up. at least here it's colocated.